sonar/modules/documents/urn.py

```diff
@@ -161,46 +173,44 @@ def register_urn_code_from_document(cls, record):
         from sonar.modules.documents.api import DocumentRecord
         from sonar.modules.documents.dnb import DnbUrnService
 
-        # TODO: verify the urn, it could be already registered
+        urn_code = DocumentRecord.get_rero_urn_code(record)
+        if not urn_code:
+            return False
+        pid = PersistentIdentifier.get(cls.urn_pid_type, urn_code)
```

Comment:
`pid` could be `None`

Reply:
No it can't an exception is raised if the pid does not exists in the db.